packages/aws-cdk-lib/aws-eks/test/cluster.test.ts

```diff
@@ -12,13 +12,13 @@ import * as kms from '../../aws-kms';
 import * as lambda from '../../aws-lambda';
 import * as cdk from '../../core';
 import * as eks from '../lib';
-import { HelmChart } from '../lib';
+import { HelmChart, NodegroupAmiType } from '../lib';
 import { KubectlProvider } from '../lib/kubectl-provider';
 import { BottleRocketImage } from '../lib/private/bottlerocket';
 
 /* eslint-disable max-len */
 
-const CLUSTER_VERSION = eks.KubernetesVersion.V1_25;
+const CLUSTER_VERSION = eks.KubernetesVersion.V1_29;
```

Comment:
If you use 1.29 here, you should provide  `kubectlLayer` for kubectl lambda layer 1.29 as well. I feel updating this file may cause some additional issues to fix and I tend to have another separate PR to do that.

If you look at my PR draft I actually created a new integ test just for AL2023 and leave the huge cluster.test.ts untouched for this PR. What do you think?

https://github.com/aws/aws-cdk/pull/29550/files#diff-fb4c143ff72c5e6ae4920d14fa54a0e15acd140583a42c768d839e46d2331ff9

Reply:
But if you can successfully deploy cluster.test.ts that's even better! All we need to do is to add `kubectlLayer` with 1.29 for the cluster creation.

Reply:
thanks for pointing out why integ test would failed, fixing.

Reply:
Hi @guessi 

Let me elaborate a little bit.

If you look at the cluster.test.ts in the main branch
https://github.com/aws/aws-cdk/blob/main/packages/aws-cdk-lib/aws-eks/test/cluster.test.ts
The current cluster version is 1.25 with NO `kubectlLayer` specified, this should pass the unit testing but in real deployment, leaving `kubectlLayer` unspecified is not recommended. I am fine if we leave it unspecified in the unit tests but if you need to update [integ.eks-cluster.ts](https://github.com/aws/aws-cdk/blob/main/packages/%40aws-cdk-testing/framework-integ/test/aws-eks/test/integ.eks-cluster.ts) and verify it with 1.29, you will need to update [this](https://github.com/aws/aws-cdk/blob/c9fb4f739f3aacf669cdd38b8431695811be5e92/packages/%40aws-cdk-testing/framework-integ/test/aws-eks/test/integ-tests-kubernetes-version.ts#L6-L14) and it would be a long huge deployment. I guess you probably will fail the integ.eks-cluster.ts somewhere. But if you are really interested to dive deep, it would be great to have integ.eks-cluster.ts to support 1.29.

In my PR. I selected another path - leave integ.eks-cluster.ts untouched and create a new integ test to make sure AL2023 would deploy with the latest k8s 1.29. It's very opinionated but I am good with both path.

Reply:
Understood!